### PR TITLE
Implement new Models

### PR DESCRIPTION
**Pull request by InterXellar**
The following Models have been implemented:
- Activity
- Presence
- Guild
- PartialGuild
- VoiceState
- VoiceRegion

### CODE REVIEW

**Comment by itsVale** [comment]
Wow, that really is a lot! Many thanks for that! I'll look at it tomorrow when I've got some time for that.

File: shitcord/models/guild.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+class PartialGuild(Model):
+    """Represents a Partial Guild Model from the Discord API
+
+    Represents an Offline Guild,
+    or a Guild whose information has not been provided through Guild Create events during the Gateway connect.
+
+    Attributes
+    ----------
+    snowflake: :class:`Snowflake`
```
Comment (itsVale) [inline shitcord/models/guild.py]: Watch out here. Having a snowflake attribute attached to an object also means that this object always has an `id` attribute which isn't documented in this case.

File: shitcord/models/guild.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        self.region = data['region']
+        self.afk_channel_id = data['afk_channel_id']
+        self.afk_timeout = data['afk_timeout']
+        self.embed_enabled = data.get('embed_enabled')
+        self.embed_channel_id = data.get('embed_channel_id')
+        self.verification_level = data['verification_level']
+        self.default_message_notifications = data['default_message_notifications']
+        self.explicit_content_filter = data['explicit_content_filter']
+        self.roles = [Role(role, http) for role in data['roles']]
+        self.emojis = [Emoji(emoji, http) for emoji in data['emojis']]
```
Comment (itsVale) [inline shitcord/models/guild.py]: Watch out here, Emoji takes 3 arguments. `[Emoji(self.id, emoji, http) for emoji in data['emojis']]` would be the way to go.

File: shitcord/models/gateway.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    DND            = 'dnd'
+    IDLE           = 'idle'
+    INVISIBLE      = 'invisible'
+    OFFLINE        = 'offline'
+
+
+class ActivityType(enum.IntEnum):
+    PLAYING   = 0
+    STREAMING = 1
+    ListENING = 2
```
Comment (200iqdevs) [inline shitcord/models/gateway.py]: I think `ListENING` is a result from PyCharm's search and replace function. Isn't it supposed to be `LISTENING`?

**Comment by InterXellar** [comment]
Hi 200iqdevs,

Yes this was an error by Pycharm
Thanks for bringing this to my attention!

Greetings InterXellar

Am Di., 15. Jan. 2019, 11:47 hat 200 IQ Devs <notifications@github.com>
geschrieben:

**Pull request merged**